lib/common/socket/evloop.c.h

```diff
@@ -115,9 +115,58 @@ static void link_to_statechanged(struct st_h2o_evloop_socket_t *sock)
     }
 }
 
-static const char *on_read_core(int fd, h2o_buffer_t **input)
+#define NSEC_PER_SEC 1000000000
+
+static inline int64_t
+timespec_to_nsec(const struct timespec *a)
+{
+       return (int64_t)a->tv_sec * NSEC_PER_SEC + a->tv_nsec;
+}
+
+#if defined(__linux__)
+static void handle_timestamp(struct st_h2o_evloop_socket_t *sock, struct msghdr *msg)
+{
+    struct scm_timestamping *ts = NULL;
+    uint64_t packet_ts = 0;
+    h2o_loop_t *loop = sock->loop;
+
+    for (struct cmsghdr *cmsg = CMSG_FIRSTHDR(msg); cmsg;
+            cmsg = CMSG_NXTHDR(msg, cmsg)) {
+
+        /* in the future we may want to harvest IP_RECVERR messages here, which could have interesting data */
+        if (cmsg->cmsg_level != SOL_SOCKET)
+            continue;
+
+        switch (cmsg->cmsg_type) {
+            case SO_TIMESTAMPNS:
+                ts = (struct scm_timestamping *)CMSG_DATA(cmsg);
```

Comment:
I believe that for SO_TIMESTAMPNS the anciliary data is a `struct timespec`. This happens to work because the address of `&ts->ts[0]` is the same as `&ts` if `ts` were a `struct timespect`

Reply:
ah yea you are right, thanks for pointing this out. i misread the docs. fixed in my latest commit.